### ISSUE TITLE
LightExpression to support managed pointers (by ref return, ref variables) and unmanaged(T*, &x)

### ISSUE THREAD

**Issue by dzmitry-lahoda** [dadhi/FastExpressionCompiler#113]
I know that Expression does not support  pointers. But I would like to implemented these in ExpressionInfo. Please vote if you think it will be useful. So issue is just placeholder to collect likes. 
As part of this process most probably I could make ExpressionInfo to support larger part of Expression (e.g. classical ref compilation). Akin to extensions proposed here #47. Probably it may be done after #66. And uncover issues with design of ExpressionInfo. Probably usage is high performant generated code which works directly on unmanaged memory(e.g. custom composed math operations, or deserializers from unmanaged memory directly) or support for layer of instrumentation when people write `ref` heavy code. Will try out how ref structs and `span` work with FEC either, also `fixed` and `stackalloc` cases probably.

**Comment by dadhi** [comment]
Currently doing a parity for ExprInfo against Expr in #66, so whatever is currently supported for Expr in FEC will be supported for FEC.LightExpression. 

We may write additional things in latter though:

```csharp
#if LIGHT_EXPRESSION
  // new stuff
#endif
```

**Comment by dzmitry-lahoda** [comment]
Ok. I am just do not have enough interest just in support only what existing in Expr, but still believe that support what not existing will got hit not yet supported for existing Expr features I may fix.

Conditional compilation is fine, so doubts exsist for enumeration of `ExpressionType`. Extending it may be `casting` int into custom int enum.

**Comment by dadhi** [comment]
Yep, the problem (or a possibily) is duslity of `ExpressionType` and types of actual expressions. E.g. many ExpressionTypes are represented by the same `BinaryExpression`.

We may reverse a `switch` to match the `Binary`, `Unary`, etc. expressions first. Then ExpressionType for new things may not be neccessary.

**Comment by dzmitry-lahoda** [comment]
Adding inspiration links (for me to look at il and probably steal `names`,  and possibly for others who may write `hpc` code using these )

https://github.com/DotNetCross/Memory.Unsafe


https://bitbucket.org/frank_niemeyer/nativeinterop/src

https://github.com/Dogwei/NewCore/blob/master/Pointer/PointerBuilder.cs

**Comment by dadhi** [comment]
The `ref return` is supported. For the rest let's tackle as a separate issue.
